### ISSUE TITLE
Error on scanning Path with \\?\p: (GetFullPathName)

### ISSUE THREAD

**Issue by Photographix** [canton7/SyncTrayzor#192]
```
Version: 1.1.0.227; Variant: Installed; Arch: Amd64
Path: C:\Program Files\SyncTrayzor\SyncTrayzor.exe
System.ArgumentException: Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch.

Parametername: path
   bei Pri.LongPath.Directory.EnumerateNormalizedFileSystemEntries(Boolean includeDirectories, Boolean includeFiles, SearchOption option, String normalizedPath, String normalizedSearchPattern)
   bei SyncTrayzor.Services.FilesystemProvider.EnumerateFiles(String path, String searchPattern, SearchOption searchOption) in C:\projects\synctrayzor\src\SyncTrayzor\Services\FilesystemProvider.cs:Zeile 83.
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 137.
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 110.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:Zeile 139.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 204.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 97.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

i am use the unicode string for long sync paths (https://msdn.microsoft.com/en-us/library/windows/desktop/aa364963(v=vs.85).aspx)
"To extend this limit to 32,767 wide characters, call the Unicode version of the function and prepend "\\\?\" to the path. "

eg.  \\\?\p:\backup\...

**Comment by canton7** [comment]
So if you open the folder in Syncthing, it starts with `\\?\p:\backup`?

Please don't do this: Syncthing is smart enough to prepend the `\\?\` when required, and C# simply can't handle it. I'll put in some logic to strip it off anyway...

**Comment by Photographix** [comment]
is a change from the previous version ? there I had the problem.

**Comment by canton7** [comment]
The filesystem watchers were probably failing to work with the previous version - they'll have fallen over, but won't have brought SyncTrayzor down (a look in the logs will confirm this, or not).

**Comment by Photographix** [comment]
the watcher Crashs
2016-01-11 13:32:13.1711 #9720 [Debug] SyncTrayzor.Services.Conflicts.ConflictFileWatcher: Starting watcher for folder: BS administration 
2016-01-11 13:32:13.1852 #9720 [Warn] SyncTrayzor.Services.FileWatcher: Watcher for \\?\p:\administration\ couldn't be created: path doesn't exist

**Comment by canton7** [comment]
Cool :smile: 

Give me a sec; I've got a build queued which should fix this.

**Comment by canton7** [comment]
In about 8 minutes there will be an intermediate build here: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.1.0.228/artifacts

Would you mind trying it, and letting me know if it fixes your issue? Thanks!

**Comment by Photographix** [comment]
yes
cool!

**Comment by canton7** [comment]
(Right, it's built)

**Comment by canton7** [comment]
@Photographix I'm planning to push out v1.1.1 in the next 20 minutes which should include this fix and others. Could you confirm whether or not the build linked above fixes the issue? Thanks!

**Comment by Photographix** [comment]
it runs 30 oder 40 seconds than Crash

Version: 1.1.0.228; Variant: Installed; Arch: Amd64
Path: C:\Program Files\SyncTrayzor\SyncTrayzor.exe
System.UnauthorizedAccessException: Zugriff verweigert

   bei Pri.LongPath.Directory.BeginFind(String normalizedPathWithSearchPattern, WIN32_FIND_DATA& findData)
   bei Pri.LongPath.Directory.<EnumerateFileSystemIterator>d__0.MoveNext()
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 137.
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 110.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:Zeile 139.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 204.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 97.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by canton7** [comment]
Mind posting the full log?

**Comment by Photographix** [comment]
2016-01-11 14:03:43.9569 #4916 [Error] SyncTrayzor.Bootstrapper: An unhandled exception occurred System.UnauthorizedAccessException Zugriff verweigert
 Pri.LongPath.SafeFindHandle BeginFind(System.String, WIN32_FIND_DATA ByRef)    bei Pri.LongPath.Directory.BeginFind(String normalizedPathWithSearchPattern, WIN32_FIND_DATA& findData)
   bei Pri.LongPath.Directory.<EnumerateFileSystemIterator>d__0.MoveNext()
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 137.
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 110.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:Zeile 139.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 204.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 97.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler) System.UnauthorizedAccessException: Zugriff verweigert

   bei Pri.LongPath.Directory.BeginFind(String normalizedPathWithSearchPattern, WIN32_FIND_DATA& findData)
   bei Pri.LongPath.Directory.<EnumerateFileSystemIterator>d__0.MoveNext()
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 137.
   bei SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:Zeile 110.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:Zeile 139.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 204.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:Zeile 97.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
2016-01-11 14:03:43.9710 #4916 [Info] SyncTrayzor.SyncThing.ApiClient.SyncThingApiClient: Requesting API shutdown 
2016-01-11 14:03:43.9710 #4916 [Debug] SyncTrayzor.SyncThing.SyncThingProcessRunner: [NLR57] 14:03:43 INFO: Shutting down 
2016-01-11 14:03:43.9850 #4916 [Info] Syncthing: [NLR57] 14:03:43 INFO: Shutting down 
2016-01-11 14:03:43.9850 #4916 [Debug] SyncTrayzor.SyncThing.SyncThingProcessRunner: [NLR57] 14:03:43 OK: Exiting 
2016-01-11 14:03:43.9850 #4916 [Info] Syncthing: [NLR57] 14:03:43 OK: Exiting

**Comment by canton7** [comment]
Thanks - I'm after the bit in the log that occurs just before the bit you pasted. It should contain something like `SyncTrayzor.Services.Conflicts.ConflictFileWatcher: Scanning folder XXX for conflict files `

**Comment by Photographix** [comment]
[SyncTrayzor.zip](https://github.com/canton7/SyncTrayzor/files/85754/SyncTrayzor.zip)

**Comment by canton7** [comment]
Cool, thanks. So it's looking in `c:\Users\tobias.bochmann` for conflict files, and is being told it can't access a sub-folder inside that. This is the same issue as #191.

In about 2 minutes there will be another build here: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.1.0.229/artifacts . Mind trying that? Even if it doesn't crash, I'm interested in the log (since it displays some more debug info).

**Comment by Photographix** [comment]
ok no problem

**Comment by canton7** [comment]
Right, that's built. Thanks for your patience :smile:

**Comment by Photographix** [comment]
[SyncTrayzor.zip](https://github.com/canton7/SyncTrayzor/files/85779/SyncTrayzor.zip)
work ;-)

**Comment by canton7** [comment]
Thanks!

Looks like you're syncing the whole of `AppData` btw - that's probably a bad idea. In particular, Bad Things will happen if you try and sync Syncthing's or SyncTrayzor's config directories. In general, applications don't like having their files changed under their feet.

**Comment by Photographix** [comment]
it is only for backup. it is not involved live in another system

**Comment by canton7** [comment]
For example, every time SyncTrayzor's log file changes, it logs about it, which leads to an endless loop of logging, telling Syncthing to refresh, logging, telling Syncthing to refresh, etc.

**Comment by Photographix** [comment]
ok then I have to rethink

**Comment by canton7** [comment]
You can try ignoring SyncTrayzor's config directory using Syncthing's ignores.

**Comment by Photographix** [comment]
thank you I'll set up in .stglobalignore. Thanks also for the quick fix!
I close the ticket?

**Comment by canton7** [comment]
Also make sure you disable filesystem watchers - that's the main thing.

Leave it open for now - I'll close it when I get v1.1.1 released.

**Comment by canton7** [comment]
Fixed in 1.1.1
